qiskit_nature/second_q/problems/electronic_structure_problem.py

```diff
@@ -77,54 +76,44 @@ def filter_criterion_custom(eigenstate, eigenvalue, aux_values):
 
     """
 
-    def __init__(
-        self,
-        driver: ElectronicStructureDriver,
-        transformers: Optional[List[BaseTransformer]] = None,
-    ):
+    def __init__(self, hamiltonian: Hamiltonian) -> None:
         """
 
         Args:
-            driver: A fermionic driver encoding the molecule information.
-            transformers: A list of transformations to be applied to the driver result.
+            hamiltonian: TODO.
+
+        Raises:
+            TypeError: TODO.
         """
-        super().__init__(driver, transformers, "ElectronicEnergy")
+        if not isinstance(hamiltonian, ElectronicEnergy):
+            raise TypeError("TODO.")
+        super().__init__(hamiltonian)
+        self.properties: ElectronicPropertiesContainer = ElectronicPropertiesContainer()
+        self.molecule: MoleculeInfo = None
+        self.basis_transform: ElectronicBasisTransform = None
+        # TODO: further refactoring:
```

Comment:
This will be part of the upcoming refactoring PR, which depends on the aforementioned integration of the `PolynomialTensor`